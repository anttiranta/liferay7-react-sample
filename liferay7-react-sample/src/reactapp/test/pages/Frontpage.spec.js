import React from 'react';
import {create} from 'react-test-renderer';
import {IntlProvider} from "react-intl";

import {useFrontPage} from "../../js/talons/useFrontPage.es";
import FrontPage from "../../js/pages/FrontPage.es";

jest.mock('../../js/talons/useFrontPage.es', () => {
    return {
        useFrontPage: jest.fn()
    }
});

const hookReturnValues = [
    {time: '15:25:43'},
    {
        refreshTime: jest.fn().mockName('refreshTime'),
        setTime: jest.fn().mockName('setTime')
    }
];

describe("FrontPage component", () => {
    beforeAll(() => {
        jest.spyOn(console, 'error').mockImplementation(() => {});
    });
    afterAll(() => {
        console.error.mockRestore();
    });

    it('renders correctly', () => {
        useFrontPage.mockReturnValueOnce(hookReturnValues);

        let instance = create(
            <IntlProvider locale="en-US">
                <FrontPage/>
            </IntlProvider>
        );

        expect(instance.toJSON()).toMatchSnapshot();
    });

    it('renders using initial values', () => {
        const myHookReturnValues = [{}, hookReturnValues[1]];
        useFrontPage.mockReturnValueOnce(myHookReturnValues);

        let instance = create(
            <IntlProvider locale="en-US">
                <FrontPage/>
            </IntlProvider>
        );

        expect(instance.toJSON()).toMatchSnapshot();
    });

    it('should call refreshTime function on button click', () => {
        useFrontPage.mockReturnValueOnce(hookReturnValues);
        const {refreshTime} = hookReturnValues[1];

        let instance = create(
            <IntlProvider locale="en-US">
                <FrontPage/>
            </IntlProvider>
        );

        const btns = instance.root.findAllByType('button');
        const refreshButton = btns[0];

        refreshButton.props.onClick();
        expect(refreshTime).toHaveBeenCalledTimes(1);
    });
});


