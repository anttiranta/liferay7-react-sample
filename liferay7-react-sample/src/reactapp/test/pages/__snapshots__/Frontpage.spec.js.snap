// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontPage component renders correctly 1`] = `
<div
  className="root"
>
  <div
    className="timeContainer"
  >
    Liferay React Sample! The time is 15:25:43
  </div>
  <div
    className="buttonContainer"
  >
    <button
      onClick={[MockFunction refreshTime]}
    >
      Refresh time from server
    </button>
  </div>
</div>
`;

exports[`FrontPage component renders using initial values 1`] = `
<div
  className="root"
>
  <div
    className="timeContainer"
  >
    Liferay React Sample! The time is unknown
  </div>
  <div
    className="buttonContainer"
  >
    <button
      onClick={[MockFunction refreshTime]}
    >
      Refresh time from server
    </button>
  </div>
</div>
`;
